### PR TITLE
FEAT : Like-History-WatchLater

### PR DESCRIPTION
**Pull request by MiheerTamkhane**
- Adding Video to Watch Later
- Like page
- Adding to History page.

### CODE REVIEW

File: src/frontend/Contexts/LikeContext.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        setLikedVideos(data.likes);
+      } catch (e) {
+        console.error(e);
+      }
+    })();
+  }, []);
+
+  const addToLikedVideosHandler = async (authToken, video) => {
+    const data = await addToLikedVideos(authToken, video);
+    console.log("folskdjfa", data);
```
Comment (gopalbbharadva) [inline src/frontend/Contexts/LikeContext.js]: remove consoles,

File: src/frontend/Pages/SingleVideoPage/SingleVideoPage.js [hunk 2]
```diff
@@ -28,6 +39,7 @@ const SingleVideoPage = () => {
       return num;
     }
   }
+  console.log("hayee", likedVideos);
```
Comment (gopalbbharadva) [inline src/frontend/Pages/SingleVideoPage/SingleVideoPage.js]: remove consoles.

File: src/frontend/Services/LikeServices/addToLikedVideos.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    const { data } = await axios.post(
+      "/api/user/likes",
+      { video },
+      {
+        headers: {
+          authorization: authToken,
+        },
+      }
+    );
+    console.log("from add to like fu", data);
```
Comment (gopalbbharadva) [inline src/frontend/Services/LikeServices/addToLikedVideos.js]: remove

File: src/frontend/Contexts/WatchLaterContext.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  }, []);
+
+  const addToWatchLaterHandler = async (authToken, video) => {
+    const data = await addToWatchLater(authToken, video);
+    setWatchLaterVideos(data);
+  };
+  const removeFromWatchLaterHandler = async (authToken, id) => {
+    const data = await removeFromWatchLater(authToken, id);
+    setWatchLaterVideos(data);
+  };
```
Comment (gopalbbharadva) [inline src/frontend/Contexts/WatchLaterContext.js:39]: Use try catch for error handling.

**Review by gopalbbharadva — APPROVED** [review]
Code looks fine, well -structured .

File: src/frontend/MyRoutes/ProtectedRoute.js [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { Navigate } from "react-router-dom";
 import { useAuth } from "../index";
 function ProtectedRoute({ children }) {
   const { auth } = useAuth();
-  return auth.status ? children : <Navigate to="/signin" replace />;
+  return auth.status ? children : <Navigate to="/join" replace />;
```
Comment (vijaytembugade) [inline src/frontend/MyRoutes/ProtectedRoute.js]: Suggestion : You can use dynamic routing as redirecting to previous url by using `state` and `location`

**Review by vijaytembugade — APPROVED** [review]
Code looks clean and code pattern is also understandable. 
Using try catch for asyn request would be better. 🔥

File: src/frontend/MyRoutes/ProtectedRoute.js (hunk 5 above)
Comment (MiheerTamkhane) [reply src/frontend/MyRoutes/ProtectedRoute.js]: yup on it. Thank you

**Pull request merged**